src/NzbDrone.Core/MediaFiles/EpisodeImport/Aggregation/Aggregators/AggregateLanguage.cs

```diff
@@ -1,71 +1,54 @@
 using System.Collections.Generic;
 using System.Linq;
 using NLog;
-using NzbDrone.Common.Extensions;
 using NzbDrone.Core.Download;
 using NzbDrone.Core.Languages;
-using NzbDrone.Core.Parser;
 using NzbDrone.Core.Parser.Model;
-using NzbDrone.Core.Tv;
+using NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.Aggregators.Augmenters.Language;
 
 namespace NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.Aggregators
 {
     public class AggregateLanguage : IAggregateLocalEpisode
     {
+        private readonly List<IAugmentLanguage> _augmentLanguages;
         private readonly Logger _logger;
 
-        public AggregateLanguage(Logger logger)
+        public AggregateLanguage(IEnumerable<IAugmentLanguage> augmentLanguages, Logger logger)
         {
+            _augmentLanguages = augmentLanguages.OrderBy(a => a.Order).ToList();
             _logger = logger;
         }
 
         public LocalEpisode Aggregate(LocalEpisode localEpisode, DownloadClientItem downloadClientItem)
         {
-            // Get languages in preferred order, download client item, folder and finally file.
-            // Non-English languages will be preferred later, in the event there is a conflict
-            // between parsed languages the more preferred item will be used.
+            var languages = new List<Language> { };
+            var languagesConfidence = Confidence.Default;
 
-            var languages = new List<Language>
-                            {
-                                GetLanguage(localEpisode.DownloadClientEpisodeInfo, localEpisode.Episodes),
-                                GetLanguage(localEpisode.FolderEpisodeInfo, localEpisode.Episodes),
-                                GetLanguage(localEpisode.FileEpisodeInfo, localEpisode.Episodes)
-                            };
-
-            var language = languages.FirstOrDefault(l => l != Language.English) ?? Language.English;
-
-            _logger.Debug("Using language: {0}", language);
-
-            localEpisode.Language = language;
+            foreach (var augmentLanguage in _augmentLanguages)
+            {
+                var augmentedLanguage = augmentLanguage.AugmentLanguage(localEpisode, downloadClientItem);
+                if (augmentedLanguage == null)
+                {
+                    continue;
+                }
 
-            return localEpisode;
-        }
+                _logger.Trace("Considering Languages {0} ({1}) from {2}", string.Join(", ", augmentedLanguage.Languages ?? new List<Language>()), augmentedLanguage.Confidence, augmentLanguage.Name);
 
-        private Language GetLanguage(ParsedEpisodeInfo parsedEpisodeInfo, List<Episode> episodes)
-        {
-            if (parsedEpisodeInfo == null)
-            {
-                return Language.English;
+                if (augmentedLanguage?.Languages != null && augmentedLanguage.Languages.Count > 0 && !(augmentedLanguage.Languages.Count == 1 && augmentedLanguage.Languages.Contains(Language.Unknown)))
```

Comment:
Here you're using: `augmentedLanguage?.Languages`, but previously in the logging line `augmentedLanguage.Languages ?? new List<Language>()` is used`, if `augmentedLanguage` wasn't previously defined it fails logging, so `augmentedLanguage?.Language` here doesn't matter and it's already continuing if `augmentedLanguage == null`. The `Languages` property should never be null (set a default value of an empty list in the constructor of that class).